### PR TITLE
Podcasting: add podcast property to publish Tracks events

### PR DESCRIPTION
**Pull request by michaeldcain**
This adds an `is_podcast_episode` property to the `calypso_editor_` Tracks events, so that we can track podcast episode publishing, scheduling, drafting, and updating.

**To test:**
- Enable podcasting by visiting Settings > Writing > Podcasting, and choosing a category
- Enable console logging of tracks events with `localStorage.setItem( 'debug', 'calypso:analytics:tracks' );`
- Publish a new post _with_ the podcasting category assigned
- Verify that the `calypso_editor_publish` Tracks event fires with the prop `is_podcast_episode: true`
- Publish a new post _without_ the podcasting category assigned
- Verify that the `calypso_editor_publish` Tracks event fires with the prop `is_podcast_episode: false`
- Repeat the previous steps with updating, scheduling, and drafting posts.
- Verify that the matching `calypso_editor_` Tracks events fire with the correct `is_podcast_episode` prop value.

### CODE REVIEW

File: client/state/posts/stats.js [hunk 1]
```diff
@@ -36,6 +37,11 @@ export const recordSaveEvent = () => ( dispatch, getState ) => {
 	const currentStatus = get( getSitePost( state, siteId, postId ), 'status', 'draft' );
 	const confirmationSidebarEnabled = isConfirmationSidebarEnabled( state, siteId );
 	const nextStatus = post.status;
+	const podcastingCategoryId = getPodcastingCategoryId( state, siteId );
+	const isPodcastEpisode =
+		podcastingCategoryId &&
+		post.terms.category &&
```
Comment (nylen) [inline client/state/posts/stats.js]: Should we check for `post.terms` too (`post.terms && post.terms.category`) ?

**Comment by nylen** [comment]
This seems fine to me, the code looks good apart from the note above.  My only other question (a question for #nosara-tracks I think) would be whether it is worth only setting this property when it is `true`, and omitting it the rest of the time (which should be the vast majority of events).

**Comment by michaeldcain** [comment]
>My only other question (a question for #nosara-tracks I think) would be whether it is worth only setting this property when it is true, and omitting it the rest of the time (which should be the vast majority of events).

@nylen: I got confirmation that they recommend to leave it to help with code clarity and internal Events tools.

**Comment by michaeldcain** [comment]
>it might make sense to make this a selector of its own (something like isPodcastEpisode( state, siteId, post ) )

@jblz: great suggestion. I'll take care of that change (and the other places it's used) in a separate PR.

**Pull request merged**